test/trajectory.jl

```diff
@@ -217,26 +221,23 @@ end
 Check whether the subtree checks adequately detect U-turns.
 """
 function check_subtree_u_turns(z0, z1, rho)
-    t = AdvancedHMC.BinaryTree(z0, z1, StrictGeneralisedNoUTurn(rho), 0, 0, 0.0)
-
-    # The left and right subtree are created in such a way that the 
-    # check_left_subtree and check_right_subtree checks should be equivalent 
-    # to the general no U-turn check.
-    tleft = AdvancedHMC.BinaryTree(z0, z0, StrictGeneralisedNoUTurn(rho - z1.r), 0, 0, 0.0)
-    tright = AdvancedHMC.BinaryTree(z1, z1, StrictGeneralisedNoUTurn(rho - z0.r), 0, 0, 0.0)
-
-    t_generalised = AdvancedHMC.BinaryTree(
-        t.zleft,
-        t.zright,
-        GeneralisedNoUTurn(t.c.rho),
-        t.sum_α,
-        t.nα,
-        t.ΔH_max
-    )
-    s1 = AdvancedHMC.isterminated(h, t_generalised)
+    tc = StrictNoUTurn()
+
+    t = AdvancedHMC.BinaryTree(z0, z1, rho, 0, 0, 0.0)
+    tleft = AdvancedHMC.BinaryTree(z0, z0, rho - z1.r, 0, 0, 0.0)
+    tright = AdvancedHMC.BinaryTree(z1, z1, rho - z0.r, 0, 0, 0.0)
+
+    # Step 0: original generalised U-turn check
+    s1 = AdvancedHMC.isterminated(tc, h, t)
 
+    # Step 1 & 2: U-turn checks for left and right subtree; see [3] for a visualisation.
+    # Step 1: Check U-turn between the leftmost phase point of `t` and 
+    #         the leftmost  phase point of `tright`, the right subtree.
     s2 = AdvancedHMC.check_left_subtree(h, t, tleft, tright)
+    # Step 2: Check U-turn between the rightmost phase point of `t` and
+    #         the rightmost phase point of `tleft`, the left subtree.
     s3 = AdvancedHMC.check_right_subtree(h, t, tleft, tright)
+    return s1 * s2 * s3
     @test s1 == s2 == s3
```

Comment:
This line won't run since the previous line has returned.

Reply:
I will remove the previous line.